plutus-core/testlib/Test/Tasty/Extras.hs

```diff
@@ -21,32 +24,47 @@ import Data.ByteString.Lazy qualified as BSL
 import Data.Text (Text)
 import Data.Text.Encoding (encodeUtf8)
 import Data.Version
-import System.FilePath ((</>))
+import System.FilePath (joinPath, (</>))
 import System.Info
 import Test.Tasty
 import Test.Tasty.Golden
 
+ghcVersion :: String
+ghcVersion = showVersion compilerVersion
+
+-- Given two lists of FilePaths, concatenate the members of the first list
+-- together, append the GHC version number, then append the concatenation of the
+-- members of the second list.
+makeVersionedFilePath :: [FilePath] -> [FilePath] -> FilePath
+makeVersionedFilePath left right = joinPath left </> ghcVersion </> joinPath right
+
 -- | A 'TestTree' of tests under some name prefix.
-type TestNested = Reader [String] TestTree
+type TestNested = Reader [FilePath] TestTree
 
 -- | Run a 'TestTree' of tests with a given name prefix.
-runTestNestedIn :: [String] -> TestNested -> TestTree
+runTestNestedIn :: [FilePath] -> TestNested -> TestTree
 runTestNestedIn path test = runReader test path
 
 -- | Run a 'TestTree' of tests with an empty prefix.
 runTestNested :: TestNested -> TestTree
 runTestNested = runTestNestedIn []
 
 -- | Descend into a name prefix.
-testNested :: String -> [TestNested] -> TestNested
+testNested :: FilePath -> [TestNested] -> TestNested
 testNested folderName =
     local (++ [folderName]) . fmap (testGroup folderName) . sequence
 
 -- | Like `testNested` but adds a subdirectory corresponding to the GHC version being used.
-testNestedGhc :: String -> [TestNested] -> TestNested
-testNestedGhc folderName = testNested (folderName </> ghcVer)
-  where
-    ghcVer = showVersion compilerVersion
+testNestedGhc :: FilePath -> [TestNested] -> TestNested
+testNestedGhc folderName = testNested (folderName </> ghcVersion)
+
+-- Create a TestTree which runs in the directory 'path/<ghc-version>/dir'
+runTestGroupNestedGhcIn :: [FilePath] -> FilePath -> [TestNested] -> TestTree
```

Comment:
I'm conflicted about whether to represent paths like `["dir1", "dir2", "dir3"]` or `"dir1/dir2/dir3".  We use both of these in the codebase, but in theory we might be in trouble on a system that doesn't use "/" as the path separator.  That's presumably very unlikely though.

Reply:
The first options seems strictly better. I'm not sure whether or not the second option works on Windows.

Reply:
> The first options seems strictly better. 

Maybe we can do that in a later PR.  You can work around it though, like [this](https://github.com/input-output-hk/plutus/blob/4886d40c75be5ef39f8f91daaed61a9b7df7c52c/plutus-core/plutus-ir/test/PlutusIR/Core/Tests.hs#L44).

Reply:
I found some tests in `plutus-ir` that had paths like `["a/b/c"]` so I changed them to `["a", "b", "c"]` while I was at it.